### PR TITLE
Feature/1.1.0 beta 4

### PR DESCRIPTION
**Pull request by katat**
- Upgrade SDK to 2.3 and integrate the off-chain balance sync check.
 
 - Even now it is integrated with SDK's off-chain sync check, I realised the time blocker is still needed to be in place because:

 1. The off-chain balance sync check is only effective when the new settlement transaction is confirmed on-chain.
 2. The transaction confirmation could take a few minutes, in which the balance sync check won't be able to prevent new payments.

So we will need two types of sync checkings. The one at the front end to check if the transaction is either mining or mined or failed. The other one is to check with the backend and the contracts to see if the off-chain balance is in sync or not.

At the moment, the time block is a simple form of checking for the front end side, as it is having a constant time for blocking. The optimal alternative would be persisting the transaction hash in localstorage, so UI can check the status of the new settlement transaction even after restarted hubii core.

### CODE REVIEW

**Comment by katat** [comment]
Updated to deprecate the time block, and check the mining status of the settlement transactions instead. The payment blocking now should be more reliable based on the checkings against both the statuses of transaction mining and the off-chain balance synchronisation.

File: src/containers/NahmiiHoc/saga.js [hunk 1]
```diff
@@ -518,6 +510,10 @@ export function* startChallenge({ stageAmount, currency, options }) {
     let errorMessage = logErrorMsg(e);
     if (errorMessage.match(/gas.*required.*exceeds/i) || errorMessage.match(/out.*of.*gas/i)) {
       errorMessage = getIntl().formatMessage({ id: 'gas_limit_too_low' });
+    } else if (errorMessage.match(/balance.*not.*synchronised/i)) {
```
Comment (morfj) [inline src/containers/NahmiiHoc/saga.js]: A style comment: Personally I find "stroustrup" style for curly braces easier to read when you have many subsequent if/elseif/else blocks. YMMV

File: src/translations/en.json [hunk 2]
```diff
@@ -262,7 +263,7 @@
     "challenge_period_progress": "There is a challenge period in progress with staging amount {staging_amount} {symbol}",
     "challenge_period_endtime": "The end time of this settlement is {endtime}. Once the challenge period is ended successfully, a dialog will be prompted for you to confirm the settlement in order to update the staged balance that is the available amount for withdrawing currency {symbol}.",
     "settling_payment": "Settling payment, waiting the transaction to be confirmed...",
-    "starting_payment_challenge": "Starting payment challenge, waiting the transaction to be confirmed...",
+    "starting_payment_challenge": "Waiting the settlement transaction to be confirmed...",
```
Comment (morfj) [inline src/translations/en.json]: _Waiting **for** the settlement transaction to be confirmed..._

File: src/containers/NahmiiHoc/saga.js (hunk 1 above)
Comment (katat) [reply src/containers/NahmiiHoc/saga.js]: Agreed. I feel the same. I will update the eslint config to allow this style.

File: src/translations/en.json (hunk 2 above)
Comment (katat) [reply src/translations/en.json]: Thanks *for* pointing out :p

**Pull request merged**